### PR TITLE
Make ver_{enable,disable}_versioning security definers

### PR DESCRIPTION
**Pull request by strk**
Only allow toggling versioning of a table to users having
usage privilege on the table owner role.

See #100

### CODE REVIEW

File: sql/01-enable_versioning.sql [hunk 1]
```diff
@@ -19,16 +20,26 @@ DECLARE
...
-      c.oid = v_table_oid
+      c.oid = p_table_oid
+    AND
+      r.oid = c.relowner
     AND
       n.oid = c.relnamespace;
 
+    -- Check that SESSION_USER is the owner of the table, or
+    -- refuse to enable versioning on this table
+    IF NOT pg_has_role(session_user, v_owner, 'usage') THEN
```
Comment (palmerj) [inline sql/01-enable_versioning.sql]: Would we want to also be nice and allow users you have the  indirect membership role to run this function as well? Or are we forcing a SET ROLE first before running the function?

File: sql/02-disable_versioning.sql [hunk 2]
```diff
@@ -1,36 +1,66 @@
...
+    WHERE
+      c.oid = p_table_oid
+    AND
+      r.oid = c.relowner
+    AND
+      n.oid = c.relnamespace;
+
+    -- Check that SESSION_USER is the owner of the table, or
+    -- refuse to enable versioning on this table
+    IF NOT pg_has_role(session_user, v_owner, 'usage') THEN
```
Comment (palmerj) [inline sql/02-disable_versioning.sql]: As comment above

File: sql/01-enable_versioning.sql (hunk 1 above)
Comment (strk) [reply sql/01-enable_versioning.sql]: What the code is doing is checking if the invoker user (SESSION_USER) has permissions to behave as if she was the OWNER of the table, that is it is the owner, or someone who got granted  the owner role or a superuser. So there should be no need to use `SET ROLE` as long as you do have permission to act in the target role name.

Comment (palmerj) [reply sql/01-enable_versioning.sql]: Ok sorry. my misunderstanding of the pg_has_role function with the 'usage' option

**Comment by strk** [comment]
Thanks for the review!

**Comment by imincik** [comment]
@strk and @palmerj thanks

**Pull request merged**